tasmota/settings.h

```diff
@@ -309,28 +309,43 @@ typedef union {                            // Restricted by MISRA-C Rule 18.4 bu
     uint32_t mirror : 1;
     uint32_t flip : 1;
     uint32_t rtsp : 1;
-    uint32_t spare4 : 1;
-    uint32_t spare5 : 1;
-    uint32_t spare6 : 1;
-    uint32_t spare7 : 1;
-    uint32_t spare8 : 1;
-    uint32_t spare9 : 1;
-    uint32_t spare10 : 1;
-    uint32_t spare11 : 1;
-    uint32_t spare12 : 1;
-    uint32_t spare13 : 1;
-    uint32_t spare14 : 1;
+    uint32_t awb : 1;
+    uint32_t awb_gain : 1;
+    uint32_t aec : 1;
+    uint32_t aec2 : 1;
+    uint32_t agc : 1;
+    uint32_t raw_gma : 1;
+    uint32_t lenc : 1;
+    uint32_t colorbar : 1;
+    uint32_t wpc : 1;
+    uint32_t dcw : 1;
+    uint32_t bpc : 1;
     uint32_t spare15 : 1;
     uint32_t spare16 : 1;
-    uint32_t spare17 : 1;
-    uint32_t spare18 : 1;
```

Comment:
This is a very high price in terms of Settings storage. Ideally I would have preferred a Berry driver complement to store all these in `_persist.json` which gives you much more flexibility. But the job is done.

Looking at the values below, most are just a single bit, or 2. Using 32 bits value for them is a complete waste. If Theo is ok to use Settings storage for these, I would request to use a bit-field instead. After a quick count, I believe they can all fit in a single 32 bits value.

Reply:
@s-hadinger They are indeed bitfields mapped inside a single uint32_t.
Same as the SetOption bitfields for example
So the total cost of the PR is 11 bit inside an already allocated uint32_t + 1 new uint32_t

Reply:
Things being said t is always questionable if a driver should always store its settings in Settings or have it initialized by a boot-rule or autoexec.bat
Most of Mi32 options for example are not saved in settings and must be re-initialiazed at each boot

Reply:
Oh this is embarrassing. I missed the fact it was already a bit field. Never look at code before having a first coffee